src/BonjourMDNSManager.cpp

```diff
@@ -464,6 +465,28 @@ class MDNSManager::PImpl
                 removeTrailingDot(rr->domain);
                 removeTrailingDot(host);
 
+                int status;
+                struct addrinfo hints;
+                struct addrinfo *result;
+                memset(&hints, 0, sizeof(hints));
+                hints.ai_family = AF_UNSPEC;
+                hints.ai_socktype = SOCK_STREAM;
+                hints.ai_flags = AI_PASSIVE;
+                if ((status = getaddrinfo(hosttarget, NULL, &hints, &result)) == 0) {
+                    if (result->ai_family == AF_INET) {
+                        char address[INET_ADDRSTRLEN];
+                        struct sockaddr_in* ipv4 = (struct sockaddr_in*)result->ai_addr;
+                        if (NULL != inet_ntop(AF_INET, &(ipv4->sin_addr), address, INET_ADDRSTRLEN)) {
+                            service.setAddress(std::move(address));
+                        }
+                    } else if (result->ai_family == AF_INET6) {
+                        char address[INET6_ADDRSTRLEN];
+                        struct sockaddr_in6* ipv6 = (struct sockaddr_in6*)result->ai_addr;
+                        if (NULL != inet_ntop(AF_INET6, &(ipv6->sin6_addr), address, INET6_ADDRSTRLEN)) {
+                            service.setAddress(std::move(address));
+                        }
+                    }
```

Comment:
`result` is a linked list, do you only want to check the first element?

Also missing `freeaddrinfo(result);`